dali/test/python/operator/test_crop.py

```diff
@@ -568,3 +567,34 @@ def test_crop_empty_layout():
     batch_size = 3
     for device in ['gpu', 'cpu']:
         yield check_crop_empty_layout, device, batch_size, in_shape
+
+
+def test_crop_arg_input():
+    @pipeline_def
+    def pipe(device, layout):
+        assert 'C' in layout
+        spatial_ndim = len(layout) - 1
+        shape = [100 if layout[i] != 'C' else 3 for i in range(len(layout))]
+        data = fn.random.uniform(range=[0, 255], shape=shape, device='cpu')
+        if device == 'gpu':
+            data = data.gpu()
+        data = fn.reshape(data, layout=layout)
+        crop_arg = fn.random.uniform(range=[10, 90], shape=(spatial_ndim,))
+        out = fn.crop(data, crop=crop_arg)
+        return out, crop_arg
+
+    def check(device, layout):
+        p = pipe(device, layout, batch_size=3, num_threads=1, device_id=0)
+        p.build()
+        out, shape = p.run()
+        ndim = len(layout)
+        channel_dim = layout.find('C')
+        spatial_dims = [k for k in range(ndim) if k != channel_dim]
+        for i in range(len(out)):
+            expected = list(np.array(shape[i], dtype=np.int32))
+            actual = [np.array(out[i].shape())[k] for k in spatial_dims]
+            assert expected == actual, f"{expected} != {actual}"
+
+    for device in ['cpu', 'gpu']:
+        for layout in ['HWC', 'FHWC', 'CHW']:
+            yield check, device, layout
```

Comment:
How about using `@params` and not abusing the test-generation as we did before? I believe the `@params` approach is way clearer, as you do not pass strage function as one of the params to `yield`...

Here's the example: https://github.com/NVIDIA/DALI/blob/0f6dba65674436ad350bf6621b6955492bebc530/dali/test/python/operator/test_remap.py#L106